src/main/java/io/vertx/core/eventbus/impl/MessageProducerImpl.java

```diff
@@ -42,6 +43,7 @@
   private int maxSize = DEFAULT_WRITE_QUEUE_MAX_SIZE;
   private int credits = DEFAULT_WRITE_QUEUE_MAX_SIZE;
   private Handler<Void> drainHandler;
+  private Handler<Void> endHandler;
```

Comment:
we should not have the `endHandler` to be a field of this class

Reply:
In order to address issue with pending queue that you described below we need to keep this field.

Reply:
drop it, we'll honor the pending queue later in another commit, it was not honoured before anyway

Reply:
I've already implemented it and made test for it. Should I rollback my last commits?
Regarding other points: I addressed it in [this comment](https://github.com/eclipse/vert.x/pull/2123#issuecomment-332031510). I would really like to know your position.